### PR TITLE
Feature/feature medication health adapters

### PR DESCRIPTION
**Pull request by xavier-campos-diaz**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public List<Medication> findMedicationsByPet(User user, Pet pet) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        List<Medication> result = new ArrayList<>();
+        List<org.pesmypetcare.usermanagerlib.datacontainers.Medication> serverMedications = new ArrayList<>();
+        serverMedications = ServiceLocator.getInstance().getMedicationManagerClient().getAllMedicationData(accessToken,
+            owner, petName);
+        for (org.pesmypetcare.usermanagerlib.datacontainers.Medication serverMedication : serverMedications) {
+            Medication med = new Medication(serverMedication);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java:97]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  13
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  12
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java  -2
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Medication.java  -8
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllPetMedications.java  -16
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  -4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMedication.java  -23
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  -3
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPetMedication.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMedication.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347&bid=17744012)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerService.java [hunk 2]
```diff
@@ -18,45 +19,50 @@
...
      * Method that updates the date of the indicated medication.
      * @param user The owner of the pet
      * @param pet The pet from which we want to update the medication
      * @param newDate The new date of the medication
      * @param oldDate The old date of the medication
+     * @param newName The new name of the medication
+     * @param oldName The old name of the medication
      */
-    void updateMedicationDate(User user, Pet pet, String newDate, String oldDate);
+    void updateMedicationKey(User user, Pet pet, String newDate, String oldDate, String newName, String oldName)
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerService.java:43]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java [hunk 3]
```diff
@@ -11,22 +12,24 @@
 
 public class PetHealthInfo {
     private Map<DateTime, Double> weight;
-    private Map<DateTime, Double> recommendedDailyKiloCalories;
+    private Map<DateTime, Double> dailyKiloCalories;
     private Map<DateTime, Integer> exerciseFrequency;
     private Map<DateTime, Integer> weeklyExercise;
     private Map<DateTime, Double> weeklyKiloCaloriesAverage;
     private Map<DateTime, Integer> washFrequency;
+    private Map<DateTime, Integer> weeklyMeals;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:20]: Codacy Issue found: [Perhaps 'weeklyMeals' could be replaced by a local variable.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:20]: Codacy Issue found: [Avoid unused private fields such as 'weeklyMeals'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        ServiceLocator.getInstance().getMedicationManagerClient().updateMedicationField(accessToken, owner, petName,
+            medication.getMedicationDate(), medication.getMedicationName(), "duration",
+            medication.getMedicationDuration());
+        ServiceLocator.getInstance().getMedicationManagerClient().updateMedicationField(accessToken, owner, petName,
+            medication.getMedicationDate(), medication.getMedicationName(), "periodicity",
+            medication.getMedicationFrequency());
+    }
+
+    @Override
+    public void updateMedicationKey(User user, Pet pet, String newDate, String oldDate, String newName,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java:51]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java [hunk 5]
```diff
@@ -57,10 +57,12 @@ public void updateMedicationBody(User user, Pet pet, Medication medication) {
     }
 
     @Override
-    public void updateMedicationDate(User user, Pet pet, String newDate, String oldDate) {
+    public void updateMedicationKey(User user, Pet pet, String newDate, String oldDate, String oldName,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java:60]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java [hunk 6]
```diff
@@ -113,10 +113,13 @@
...
      * @param pet The pet from which we want to update the medication
      * @param medication The updated medication
-     * @param newDate The new date of the meal
+     * @param newDate The new date of the medication
      * @param updatesDate True if the date has to be updated or false otherwise
+     * @param newName The new name of the medication
+     * @param updatesName True if the name has to be updated or false otherwise
      */
-    void updatePetMedication(Pet pet, Medication medication, String newDate, boolean updatesDate);
+    void updatePetMedication(Pet pet, Medication medication, String newDate, boolean updatesDate, String newName,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java:121]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java [hunk 7]
```diff
@@ -257,10 +270,34 @@ public double getWeeklyKiloCalAverageForDate(DateTime date) {
...
+            this.weeklyKiloCaloriesAverage.put(date, kcal);
+            this.weeklyMeals.put(date, 1);
+        }*/
+    }
+
+    /**
+     * Obtains the monday of the indicated day.
+     * @param date The date for which we want to find the monday
+     */
+    private void obtainDateMonday(DateTime date) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:291]: Codacy Issue found: [Avoid unused private methods such as 'obtainDateMonday(DateTime)'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

**Review by enrichp — APPROVED** [review]

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        DateTime oldDateTime = DateTime.Builder.buildFullString(oldDate);
+        MedicationData medicationData = ServiceLocator.getInstance().getMedicationManagerClient().getMedicationData(
+            accessToken, owner, petName, oldDateTime, oldName);
+        Medication currentMedication = new Medication(oldName, medicationData.getQuantity(),
+            medicationData.getPeriodicity(), medicationData.getDuration(), oldDateTime);
+        deleteMedication(user, pet, currentMedication);
+        System.out.println("New name : " + newName + " new date " + newDate);
```
Comment (danicm47) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java]: Delete this print

**Review by danicm47 — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  13
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  12
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  -4
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  -3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllPetMedications.java  -16
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Medication.java  -8
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMedication.java  -23
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java  -2
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPetMedication.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMedication.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347&bid=17744012)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        ServiceLocator.getInstance().getMedicationManagerClient().updateMedicationField(accessToken, owner, petName,
+            medication.getMedicationDate(), medication.getMedicationName(), "duration",
+            medication.getMedicationDuration());
+        ServiceLocator.getInstance().getMedicationManagerClient().updateMedicationField(accessToken, owner, petName,
+            medication.getMedicationDate(), medication.getMedicationName(), "periodicity",
+            medication.getMedicationFrequency());
+    }
+
+    @Override
+    public void updateMedicationKey(User user, Pet pet, String newDate, String oldDate, String newName,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java:51]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java (hunk 5 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/StubMedicationService.java:60]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java (hunk 3 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:20]: Codacy Issue found: [Perhaps 'weeklyMeals' could be replaced by a local variable.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java (hunk 7 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:291]: Codacy Issue found: [Avoid unused private methods such as 'obtainDateMonday(DateTime)'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerService.java (hunk 2 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerService.java:43]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java (hunk 6 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java:121]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java (hunk 3 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java:20]: Codacy Issue found: [Avoid unused private fields such as 'weeklyMeals'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    public List<Medication> findMedicationsByPet(User user, Pet pet) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        List<Medication> result = new ArrayList<>();
+        List<org.pesmypetcare.usermanagerlib.datacontainers.Medication> serverMedications = new ArrayList<>();
+        serverMedications = ServiceLocator.getInstance().getMedicationManagerClient().getAllMedicationData(accessToken,
+            owner, petName);
+        for (org.pesmypetcare.usermanagerlib.datacontainers.Medication serverMedication : serverMedications) {
+            Medication med = new Medication(serverMedication);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MedicationManagerAdapter.java:96]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5412347)

**Pull request merged**